### PR TITLE
Add CI CD workflow file

### PR DESCRIPTION
**Pull request by MoriaYechiel**

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        run: |
+          cd server
+          sbt test
+
+# Copy logs to 1 directory, since upload-artifact doesnt support multiple directories or regular expressions
+      - name: Save logs
+        if: always()
+        run: |
+          mkdir logsArtifacts
+          find . -name logs | xargs -I LOG cp -r --parents LOG logsArtifacts/
```
Comment (e-orz) [inline .github/workflows/workflow.yml]: Why is this find needed?

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+#####################################
+##        Save Artifacts           ##
+#####################################
+      - uses: actions/upload-artifact@master
```
Comment (e-orz) [inline .github/workflows/workflow.yml]: lock it to a version

File: .github/workflows/workflow.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        if: always()
+        with:
+          name: logsArtifacts
+          path: logsArtifacts
+      - uses: actions/upload-artifact@master
```
Comment (e-orz) [inline .github/workflows/workflow.yml]: same

File: .github/workflows/workflow.yml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        if: always()
+        with:
+          name: client1.out
+          path: server/cmwell-grid/target/client1.out
+      - uses: actions/upload-artifact@master
```
Comment (e-orz) [inline .github/workflows/workflow.yml]: same

File: .github/workflows/workflow.yml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        if: always()
+        with:
+          name: client2.out
+          path: server/cmwell-grid/target/client2.out
+      - uses: actions/upload-artifact@master
```
Comment (e-orz) [inline .github/workflows/workflow.yml]: same

File: .github/workflows/workflow.yml [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        if: always()
+        with:
+          name: node.out
+          path: server/cmwell-grid/target/node.out
+
+#####################################
+##  Running all integration tests  ##
+##################################### 
+  integrationTests:
+    runs-on: ubuntu-latest
```
Comment (e-orz) [inline .github/workflows/workflow.yml]: lock it to a version 18.04

File: .github/workflows/workflow.yml [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    steps:
+    - uses: actions/checkout@v1
+    - name: Install Scala
+      run: |
+        wget https://downloads.lightbend.com/scala/2.12.8/scala-2.12.8.tgz
+        sudo tar -xf scala-2.12.8.tgz
+        sudo ln -s `pwd`/scala-2.12.8/bin/scala /usr/local/bin/scala
+    - name: Test
+      run: |
+        cd server
```
Comment (e-orz) [inline .github/workflows/workflow.yml]: working-directory?

File: .github/workflows/workflow.yml [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Install Scala
+      run: |
+        wget https://downloads.lightbend.com/scala/2.12.8/scala-2.12.8.tgz
+        sudo tar -xf scala-2.12.8.tgz
+        sudo ln -s `pwd`/scala-2.12.8/bin/scala /usr/local/bin/scala
+    - name: Test
```
Comment (e-orz) [inline .github/workflows/workflow.yml]: "Intergration Tests"

File: .github/workflows/workflow.yml [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+      run: |
+        cd server
+        sbt it:test
```
Comment (e-orz) [inline .github/workflows/workflow.yml]: change to `IntegrationTests / test`

File: .github/workflows/workflow.yml [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+        
+#####################################
+##       Save IT Artifacts         ##
+##################################### 
+    - uses: actions/upload-artifact@master
```
Comment (e-orz) [inline .github/workflows/workflow.yml]: same

File: .github/workflows/workflow.yml [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+      if: always()
+      with:
+        name: itLogs
+        path: server/cmwell-it/target/logs
+
+    - uses: actions/upload-artifact@master
```
Comment (e-orz) [inline .github/workflows/workflow.yml]: same

File: .github/workflows/workflow.yml (hunk 1 above)
Comment (MoriaYechiel) [reply .github/workflows/workflow.yml]: https://github.com/CM-Well/CM-Well/commit/2efe933e88d11dac19995268724cdbcda16d46fa/checks?check_suite_id=248257504

Comment (MoriaYechiel) [reply .github/workflows/workflow.yml]: https://github.com/CM-Well/CM-Well/commit/aaa57a1ccea112fddd89b214741e30ab35154697/checks?check_suite_id=248119367#step:5:6

**Pull request merged**